src/app/clusters/thread-network-directory-server/DefaultThreadNetworkDirectoryStorage.cpp

```diff
@@ -46,7 +46,8 @@ CHIP_ERROR DefaultThreadNetworkDirectoryStorage::StoreIndex()
 {
     VerifyOrDie(mInitialized);
     StorageKeyName key = DefaultStorageKeyAllocator::ThreadNetworkDirectoryIndex();
-    return mStorage.SyncSetKeyValue(key.KeyName(), mExtendedPanIds, mCount * ExtendedPanId::size());
+    static_assert(kCapacity * ExtendedPanId::size() <= UINT16_MAX); // kCapacity >= mCount
+    return mStorage.SyncSetKeyValue(key.KeyName(), mExtendedPanIds, static_cast<uint16_t>(mCount * ExtendedPanId::size()));
```

Comment:
See line 33:

```
    SuccessOrExit(err = mStorage.SyncGetKeyValue(key.KeyName(), mExtendedPanIds, size));
    VerifyOrExit(size % ExtendedPanId::size() == 0, err = CHIP_ERROR_INTERNAL);
```

You can get in a situation where the list is unloadable due to changing size, if `sizeof(mExtendedPanIds)` changes.

- Suggest having a separate storage key to maintain the count, so that it is possible to ensure buffers are based on the actual number of elements at play.
- It looks like there is no way to clear the index and the overall storage, especially when last fabric is removed from the device. Please register a FabricTable::Delegate at the cluster level to handle the removal of all TBR data on last fabric removed.

Reply:
The size of ExtendedPanID won’t change because it’s mandated by the Thread spec. This change is just to avoid a warning that popped up on some compiler versions when building the THNetDir cluster in all-clusters, so it’s not strictly needed now but seemed useful enough to leave in.

Good point about clearing the table on removal of the last fabric, I’ll address that in a separate PR. We may also need to add that to the spec.

Reply:
Filed #34750